extensions/guacamole-auth-restrict/src/main/resources/controllers/timeRestrictionFieldController.js

```diff
@@ -0,0 +1,302 @@
+                        if (origDay <= 0) 
+                            entry.weekDay = '' + 6;
+                        else
+                            entry.weekDay = '' + (--origDay);
+
+                    }
+
+                    // If UTC day is less than local day, we add a day, 
+                    // wrapping as required.
+                    else if (entry.startTime.getDay() > entry.startTime.getUTCDay()) {
+                        if (origDay >= 6)
+                            entry.weekDay = '' + 0;
+                        else
+                            entry.weekDay = '' + (++origDay);
+                    }
+
+                    // Local day and UTC day are the same, adjust the display day
+                    else
+                        entry.weekDay = '' + origDay;
+
+                }
+
+                restrictions.push(entry);
+            }
+        }
+        
+        return restrictions;
+        
+    };
+    
+    /**
+     * Parse the restrictions in the field into a string that can be stored
+     * in an underlying module.
+     * 
+     * @param {TimeRestrictionEntry[]} restrictions
+     *     The array of restrictions that will be converted to a string.
+     * 
+     * @returns {String}
+     *     The string containing the restriction data that can be stored in e.g.
+     *     a database.
+     */
+    const storeRestrictions = function storeRestrictions(restrictions) {
+        
+        // If there are no members of the array, just return an empty string.
+        if (restrictions === null || restrictions.length < 1)
+            return '';
+        
+        let restrString = '';
+        for (let i = 0; i < restrictions.length; i++) {
+            // If any of the properties are not defined, skip this one.
+            if (!Object.hasOwn(restrictions[i], 'weekDay')
+                    || restrictions[i].weekDay === null
+                    || restrictions[i].weekDay === ''
+                    || !Object.hasOwn(restrictions[i], 'startTime')
+                    || restrictions[i].startTime === null
+                    || !(restrictions[i].startTime instanceof Date)
+                    || !Object.hasOwn(restrictions[i], 'endTime') 
+                    || restrictions[i].endTime === null
+                    || !(restrictions[i].endTime instanceof Date))
+                continue;
+            
+            // If this is not the first item, then add a semi-colon separator
+            if (restrString.length > 0)
+                restrString += ';';
+            
+            // When these fields first gets a value, the default year is 1970
+            // In order to avoid issues with Daylight Savings Time, we have to
+            // work around this.
+            if (restrictions[i].startTime instanceof Date && restrictions[i].startTime.getFullYear() === 1970) {
+                let startHour = restrictions[i].startTime.getHours();
```

Comment:
I'd consider moving this date fixing stuff into a shared function (for the start and end times).